File: java/src/test/java/problems101_200/ConstructBinaryTreeFromPreorderAndInorderTraversalTest.java
package problems101_200;

import concepts.TreeNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ConstructBinaryTreeFromPreorderAndInorderTraversalTest extends SampledTest {
    @Test
    public void test() {
        for (ConstructBinaryTreeFromPreorderAndInorderTraversal solution
            : ConstructBinaryTreeFromPreorderAndInorderTraversal.values()) {
            for (ConstructBinaryTreeFromPreorderAndInorderTraversalSample sample
                : getSamples(ConstructBinaryTreeFromPreorderAndInorderTraversalSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.buildTree(sample.input.preorder, sample.input.inorder))
                    .isEqualTo(TreeNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/ConstructBinaryTreeFromPreorderAndInorderTraversalTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ConstructBinaryTreeFromPreorderAndInorderTraversalTest : SampledTest() {
    @Test
    fun test() =
        ConstructBinaryTreeFromPreorderAndInorderTraversal.entries.forEach { solution ->
            getSamples<Array<ConstructBinaryTreeFromPreorderAndInorderTraversalSample>>()
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.buildTree(input.preorder, input.inorder))
                        .isEqualTo(TreeNode.of(*output))
                }
        }
}
